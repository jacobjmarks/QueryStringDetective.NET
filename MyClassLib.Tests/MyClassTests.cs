using System.Text.Json;
using Snapshooter;

namespace MyClassLib.Tests;

public class MyClassTests
{
    [Theory]
    [InlineData("foo")]
    [InlineData("a&q=b&q=c")]
    [InlineData("null")]
    [InlineData("0")]
    [InlineData("1")]
    [InlineData("-1")]
    [InlineData("1&q=2&q=3")]
    [InlineData("1&q=2&q=.3")]
    [InlineData("-1&q=0&q=1")]
    [InlineData("2147483647")]
    [InlineData("-2147483648")]
    [InlineData("9223372036854775807")]
    [InlineData("-9223372036854775808")]
    [InlineData("18446744073709551615")]
    public async Task AssertBinding(string queryString)
    {
        var results = await MyClass.TestQueryStringBindingAsync("?q=" + queryString);
        var assertions = results.Where(r => !r.IsErroneous).Select(r => r.ToString());
        assertions.Should().MatchSnapshot(SnapshotNameExtension.Create(ToValidFileName(queryString)));
    }

    private static string ToValidFileName(string value)
    {
        foreach (var invalidChar in Path.GetInvalidFileNameChars())
            value = value.Replace(invalidChar, '_');

        return value;
    }
}