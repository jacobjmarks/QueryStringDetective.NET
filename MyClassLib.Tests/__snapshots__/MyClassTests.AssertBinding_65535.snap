Input: 65535

Bindings:
string     "65535"
string[]   ["65535"]
int        65535
int[]      [65535]
double     65535
double[]   [65535]
float      65535
float[]    [65535]
decimal    65535
decimal[]  [65535]
uint       65535
uint[]     [65535]
long       65535
long[]     [65535]
ulong      65535
ulong[]    [65535]
ushort     65535
ushort[]   [65535]

Erroneous:
char       400 Bad Request
char[]     400 Bad Request
bool       400 Bad Request
bool[]     400 Bad Request
sbyte      400 Bad Request
sbyte[]    400 Bad Request
short      400 Bad Request
short[]    400 Bad Request
byte       400 Bad Request
byte[]     400 Bad Request
